emcc.py

```diff
@@ -2190,10 +2190,13 @@ def check_memory_setting(setting):
       # LTO these symbols don't exist prior the linking.
       '___cxa_is_pointer_type',
       '___cxa_can_catch',
-
-      # Emscripten exception handling can generate invoke calls, and they call
-      # setThrew(). We cannot handle this using deps_info as the invokes are not
-      # emitted because of library function usage, but by codegen itself.
+    ]
+  if not settings.DISABLE_EXCEPTION_CATCHING or settings.SUPPORT_LONGJMP:
+    settings.EXPORTED_FUNCTIONS += [
+      # Emscripten exception handling and setjmp/longjmp handling can generate
+      # invoke calls, and they call setThrew(). We cannot handle this using
+      # deps_info as the invokes are not emitted because of library function
+      # usage, but by codegen itself.
       '_setThrew',
```

Comment:
I think the problem here is that you are introducing a new included-by-default function (since SUPPORT_LONGJMP defaults to 1).. so this will regress code size tests for tiny programs I believe.    If a program never calls longjmp it should never need `_setThrew`...   for exception handling the user has opted in so I more reasonable to accept that cost but most user don't explicitly opt into longjmp (they just get it because its the default).

Reply:
Removed it and added deps in deps_info.py instead with comments.